### PR TITLE
slasher server

### PR DESCRIPTION
**Pull request by shayzluf**
[Part of]  #3133

---

# Description

**Write why you are making the changes in this pull request**
separate slasher server

### CODE REVIEW

**Comment by nisdas** [comment]
@shayzluf is this ready for review ?

**Comment by shayzluf** [comment]
Yes

File: slasher/rpc/server.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+// providing RPC endpoints for retrieving slashing proofs for malicious validators.
+type Server struct {
+	SlasherDb *db.Store
+	ctx       context.Context
+}
+
+// IsSlashableAttestation returns an attester slashing if the attestation submitted
+// is a slashable vote.
+func (ss *Server) IsSlashableAttestation(ctx context.Context, req *ethpb.Attestation) (*ethpb.AttesterSlashing, error) {
+	return nil, status.Error(codes.Unimplemented, "not implemented")
```
Comment (terencechain) [inline slasher/rpc/server.go]: can you add a todo for this "not implemented"?

File: slasher/rpc/server.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+		if err != nil {
+			return nil, err
+		}
+	}
+	return pSlashingsResponse, nil
+}
+
+// SlashableProposals is a subscription to receive all slashable proposer slashing events found by the watchtower.
+func (ss *Server) SlashableProposals(req *types.Empty, server ethpb.Slasher_SlashableProposalsServer) error {
+	return status.Error(codes.Unimplemented, "not implemented")
```
Comment (terencechain) [inline slasher/rpc/server.go]: same, could be the same issue here

**Pull request merged**